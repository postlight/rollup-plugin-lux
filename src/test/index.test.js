// @flow
import path from 'path';

import fs from 'fs-promise';

import plugin from '../index';

describe('#transform()', () => {
  const appPath = path.posix.join(__dirname, '__fixtures__');

  describe('with lux sub class', () => {
    describe('[ClassDeclaration] export default [Identifier]', () => {
      let id;
      let subject;

      beforeEach(() => {
        id = path.join(
          __dirname,
          '__fixtures__',
          'export-default-statement-lux-sub-class.js'
        );
        return fs
          .readFile(id, 'utf8')
          .then(data => {
            subject = data;
          });
      });

      it('appends an Object.defineProperty call to the module', () => {
        expect(plugin(appPath).transform(subject, id)).toMatchSnapshot();
      });
    });

    describe('export default [ClassExpression]', () => {
      let id;
      let subject;

      beforeEach(() => {
        id = path.join(
          __dirname,
          '__fixtures__',
          'export-default-expression-lux-sub-class.js'
        );
        return fs
          .readFile(id, 'utf8')
          .then(data => {
            subject = data;
          });
      });

      it('appends an Object.defineProperty call to the module', () => {
        expect(plugin(appPath).transform(subject, id)).toMatchSnapshot();
      });
    });
  });

  describe('without lux sub class', () => {
    describe('[ClassDeclaration] export default [Identifier]', () => {
      let id;
      let subject;

      beforeEach(() => {
        id = path.join(
          __dirname,
          '__fixtures__',
          'export-default-statement-generic-class.js'
        );
        return fs
          .readFile(id, 'utf8')
          .then(data => {
            subject = data;
          });
      });

      it('does not modify the source', () => {
        expect(plugin(appPath).transform(subject, id)).toMatchSnapshot();
      });
    });

    describe('[FunctionDeclaration] export default [Identifier]', () => {
      let id;
      let subject;

      beforeEach(() => {
        id = path.join(__dirname, '__fixtures__', 'export-default-function.js');
        return fs
          .readFile(id, 'utf8')
          .then(data => {
            subject = data;
          });
      });

      it('does not modify the source', () => {
        expect(plugin(appPath).transform(subject, id)).toMatchSnapshot();
      });
    });

    describe('export default [ArrowFunctionExpression]', () => {
      let id;
      let subject;

      beforeEach(() => {
        id = path.join(
          __dirname,
          '__fixtures__',
          'export-default-arrow-function.js'
        );
        return fs
          .readFile(id, 'utf8')
          .then(data => {
            subject = data;
          });
      });

      it('does not modify the source', () => {
        expect(plugin(appPath).transform(subject, id)).toMatchSnapshot();
      });
    });

    describe('named export', () => {
      let id;
      let subject;

      beforeEach(() => {
        id = path.join(__dirname, '__fixtures__', 'named-export.js');
        return fs
          .readFile(id, 'utf8')
          .then(data => {
            subject = data;
          });
      });

      it('does not modify the source', () => {
        expect(plugin(appPath).transform(subject, id)).toMatchSnapshot();
      });
    });
  });
});
