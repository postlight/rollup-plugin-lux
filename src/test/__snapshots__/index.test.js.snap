exports[`#transform() with lux sub class [ClassDeclaration] export default [Identifier] appends an Object.defineProperty call to the module 1`] = `
Object {
  "code": "import { Model } from \'lux-framework\';

class User extends Model {

}

Object.defineProperty(User, \'name\', { value: \'User\' });

export default User;
",
  "map": Object {
    "file": null,
    "mappings": "AAAA;;;;;;uDAIC;;;",
    "names": Array [],
    "sources": Array [
      null,
    ],
    "sourcesContent": Array [
      null,
    ],
    "version": 3,
  },
}
`;

exports[`#transform() with lux sub class export default [ClassExpression] appends an Object.defineProperty call to the module 1`] = `
Object {
  "code": "import { Model } from \'lux-framework\';

export default class User extends Model {

}

Object.defineProperty(User, \'name\', { value: \'User\' });
",
  "map": Object {
    "file": null,
    "mappings": "AAAA;;;;;;uDAIC;",
    "names": Array [],
    "sources": Array [
      null,
    ],
    "sourcesContent": Array [
      null,
    ],
    "version": 3,
  },
}
`;

exports[`#transform() without lux sub class [ClassDeclaration] export default [Identifier] does not modify the source 1`] = `
Object {
  "code": "class Point {
  constructor(x, y) {
    Object.assign(this, { x, y });
  }
}

Object.defineProperty(Point, \'name\', { value: \'Point\' });

export default Point;
",
  "map": Object {
    "file": null,
    "mappings": "AAAA;;;;;;yDAIC;;;",
    "names": Array [],
    "sources": Array [
      null,
    ],
    "sourcesContent": Array [
      null,
    ],
    "version": 3,
  },
}
`;

exports[`#transform() without lux sub class [FunctionDeclaration] export default [Identifier] does not modify the source 1`] = `
Object {
  "code": "function test() {

}

export default test;
",
  "map": Object {
    "file": null,
    "mappings": "AAAA;;;;;",
    "names": Array [],
    "sources": Array [
      null,
    ],
    "sourcesContent": Array [
      null,
    ],
    "version": 3,
  },
}
`;

exports[`#transform() without lux sub class export default [ArrowFunctionExpression] does not modify the source 1`] = `
Object {
  "code": "export default () => undefined;
",
  "map": Object {
    "file": null,
    "mappings": "AAAA;",
    "names": Array [],
    "sources": Array [
      null,
    ],
    "sourcesContent": Array [
      null,
    ],
    "version": 3,
  },
}
`;

exports[`#transform() without lux sub class named export does not modify the source 1`] = `
Object {
  "code": "export const TEST = true;
",
  "map": Object {
    "file": null,
    "mappings": "AAAA;",
    "names": Array [],
    "sources": Array [
      null,
    ],
    "sourcesContent": Array [
      null,
    ],
    "version": 3,
  },
}
`;
